clang/include/clang/Basic/DiagnosticGroups.td

```diff
@@ -1239,6 +1239,7 @@ def Sycl2020Compat : DiagGroup<"sycl-2020-compat">;
 def SyclStrict : DiagGroup<"sycl-strict", [ Sycl2017Compat, Sycl2020Compat]>;
 def SyclTarget : DiagGroup<"sycl-target">;
 def SyclFPGAMismatch : DiagGroup<"sycl-fpga-mismatch">;
+def SyclCompat : DiagGroup<"sycl-compat">;
```

Comment:
I'm on the fence on the naming of the `-W` option here.

Reply:
Not up to me, but I agree here.  I don't think this warning group fits well here.  I'd suggest the CFE/driver reviewers see if we can bikeshed this better

Reply:
I'm not sure what the group should be named here. `InvalidSourceFile`/`UnsupportedFileType`/`LanguageMismatch`? I don't really like any of these. @premanandrao do you have a suggestion?

Reply:
> I did quick search to see if I can find a better Warning group and I saw 2 other diagnostics which look related -
> 
> ```
> def err_drv_fsycl_with_c_type : Error<
>   "'%0' must not be used in conjunction with '-fsycl', which expects C++ source">;
> 
> def warn_drv_treating_input_as_cxx : Warning<
>   "treating '%0' input as '%1' when in C++ mode, this behavior is deprecated">,
>   InGroup<Deprecated>;
> ```
> 
> I assume the latter is used for non-sycl programs but would you know what the former is used for now?

Hi @elizabethandrews , In above 2 defs "err_drv_fsycl_with_c_type" can only be used for error message. so I can't use it for a warning. And the other one "warn_drv_treating_input_as_cxx" I tried using it but @mdtoguchi said "The warning is not accurate for -fsycl". So I have created new one.

Reply:
Sorry for not being clear.  My question was - when is `err_drv_fsycl_with_c_type ` thrown ? I was curious about warning vs error functionality here.

Reply:
Right, but what is the scenario in which error is thrown? If we are converting C types to C++ types for SYCL, why does that error diagnostic exist.

Reply:
This error emits when 'c' compilation is forced in -fsycl mode.

Reply:
@hchilama I'm ok with `InvalidSourceFile` or `ExpectedFileType`. If you agree, please change the group name. Thanks!

Reply:
Sure thanks @elizabethandrews, I will update the group name.